submissions/ses3332002/memory-game/script.js

```diff
@@ -0,0 +1,152 @@
+      item.disabled = !item.disabled;
+      item.classList.toggle("form_item-desabled");
+    });
+  };
+  
+  function checkOrientationAndSize() {
+    if (document.documentElement.clientHeight > document.documentElement.clientWidth) {
+      if (sizeSelector.value == gamefieldSize1) {
+        gameArea.classList.add("game_area-3columns");
+      } else {
+        gameArea.classList.add("game_area-4columns");
+      };
+    } else {
+      if (sizeSelector.value <= gamefieldSize2) {
+        gameArea.classList.add("game_area-4columns");
+      } else if (sizeSelector.value == gamefieldSize3){
+        gameArea.classList.add("game_area-5columns");
+      } else {
+        gameArea.classList.add("game_area-6columns");
+      };
+    };
+  };
+  
+  function shuffle(array) {
+    for (let i = array.length - 1; i > 0; i--) {
+      const j = Math.floor(Math.random() * (i + 1));
+      [array[i], array[j]] = [array[j], array[i]];
+    };
+  };
+  
+  function prepareGameField() {
+    numbersArray = [];
+    for (let i=0; i < sizeSelector.value/2; i++) {
+      numbersArray[i] = i;
+    };
+    numbersArray = numbersArray.concat(numbersArray);
+    
+    shuffle(numbersArray);
+    checkOrientationAndSize();
+    
+    const fragment = document.createDocumentFragment();
+    for (let i=0; i < sizeSelector.value; i++) {
+      let card = document.createElement("button");
+      card.classList.add("card");
+      card.innerHTML = `<div class="card__front" style="background-image: url(img/${cardThemeSelected.value}_bg.jpg);"></div><div class="card__back" style="background-image: url(img/${cardThemeSelected.value}_${numbersArray[i] + 1}.jpg);"></div>`;
+      fragment.append(card);
+    };
+    gameArea.append(fragment);
+    gameArea.addEventListener("click", checkCard);
+    cardsArray = Array.from(document.querySelectorAll(".card"));
+  };
+  
+  function checkCard({target}) {
+    if (target.className != "card") {
+      return;
+    };
+    
+    flipCard.call(target);
+    setTimeout(function() {
+      if (!cardSelected) {
+        cardSelected = cardsArray.indexOf(target) + 1;
+        lockCard.call(target);
+      } else if ((numbersArray[cardSelected - 1] == numbersArray[cardsArray.indexOf(target)])&&(cardSelected - 1 != cardsArray.indexOf(target))) {
+        vanishCard.call(cardsArray[cardSelected - 1]);
+        vanishCard.call(target);
+        cardSelected = "";
+        matchedCounter++;
+        if (matchedCounter == sizeSelector.value/2) {
+          hailWinner();
+        };
+      } else {
+        flipCard.call(cardsArray[cardSelected - 1]);
+        unlockCard.call(cardsArray[cardSelected - 1])
+        cardSelected = "";      
+        flipCard.call(target);
+      };
+    }, animationDelay);
+  };
+  
+  function vanishCard() {
+    this.classList.add("card-vanish");
+    this.setAttribute("tabindex", "-1");
+  };
+  
+  function flipCard() {
+    this.firstElementChild.classList.toggle("card__front-flip");
```

Comment:
It's better to use `this.closest('someSelector')...`. In this way, you don't depend on your layout structure and just check child by it's className (or by another query)

Reply:
I remade it.